app/assets/javascripts/govuk_publishing_components/lib/single-consent-functions.js

```diff
@@ -0,0 +1,65 @@
+/* global GovSingleConsent */
+// = require govuk-single-consent/dist/singleconsent.iife.js
+
+(function (root) {
+  'use strict'
+  window.GOVUK = window.GOVUK || {}
+
+  window.GOVUK.singleConsent = {
+    init: function (callback) {
+      if (!this.useConsentApi()) {
+        return
+      }
+      callback = callback || this.apiCallBack
+      // determine where we are and set the consent api URL accordingly
+      if (!this.url) {
+        this.url = 'staging'
+        var environment = window.GOVUK.loadAnalytics.getEnvironment(window.GOVUK.analyticsGa4.core.trackFunctions.getHostname())
```

Comment:
Yeah, I've been wondering about this. It started off as a small change and it's snowballed a bit. I think this is what I'd like to do:

- take the list of `domains` out of `load-analytics` and put it somewhere else, like a whole new file
- include the `loadExtraDomains` function with that
- leave the rest of the file as-is

Does that sound sensible?

Reply:
Sounds good - regarding a new file, I'm guessing this would be another JS script to load into the DOM as well? This is because I believe `loadAnalytics` always loads before `application.js`, so we might be stuck trying to solve this as you would probably need these domains to exist before `loadAnalytics` is called.

Reply:
Have come up with a solution, hopefully is sensible and understandable, see newest commit.

Reply:
@andysellick Nice, looks good 👍 Do you think our tests should ensure the `production` values always stay the same in that object, or is that overkill? I guess there's a risk that the production values change without us knowing and that affects analytics/cookie consent.

Reply:
It's a good thought, but I think it might be overkill. Two possible things I think you could be talking about here...

First, that the production details would be somehow lost during the `loadExtraDomains` function. This is semi-tested, in that the test first gets the hard coded domains, then appends the test data onto it, then checks that the output matches this. I guess we could hard code the expected production values (like you said) for `name` into the test, and check that, except that regardless of the environment, if we're passing extra values, the intent is that these extra values are used, not the defaults - so if the original production values are lost somehow, it shouldn't matter (unless we pass extra values on production, but that'd be strange).

Second, that someone makes a code change that alters the production values. I don't think we normally check that variables like this remain the same, and I'd hope that any accidental change to these values would be picked up in review (think you meant the first thing, but just covering all the options).

Reply:
No worries - I was referring to the second thing 👍